### PR TITLE
add option to pass attributes to unauthorized replies

### PR DESCRIPTION
**Pull request by maxbeatty**
This addition allows someone to optionally specify an attributes object to pass to [Boom.unauthorized](https://github.com/hapijs/boom#boomunauthorizedmessage-scheme-attributes).

Some backstory: AWS SNS allows subscriptions from HTTPS endpoints with basic authentication. Their confirmation flow involves an initial unauthenticated request to the endpoint followed by an authenticated request. I wasn't receiving the second authenticated request because the `WWW-Authentication` header didn't specify a realm (e.g. `WWW-Authentication: Basic realm="hapi"`). Thankfully, `Boom.unauthorized` made this easy to add.

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -741,6 +741,42 @@ it('accepts request object in validateFunc', (done) => {
     });
 });
 
+it('includes additional attributes in WWW-Authenticate header', (done) => {
+    const server = new Hapi.Server();
```
Comment (mtharrison) [inline test/index.js]: Missing blank line here

File: test/index.js [hunk 2]
```diff
@@ -741,6 +741,42 @@ it('accepts request object in validateFunc', (done) => {
...
+
+                return reply('ok');
+            },
+            config: {
+                auth: 'default'
+            }
+        });
+
+        const request = { method: 'POST', url: '/', headers: {
+            // intentionally empty
```
Comment (mtharrison) [inline test/index.js]: Why include this empty object?

**Comment by mtharrison** [comment]
Thanks for the contribution. Looks good apart from those couple of comments. Part of me wonders though if this will ever be used for anything apart from setting the `realm` and if it'd be clearer to just have a `realm` option. We'll probably just go with this though for now, it's simple enough to use and the README mentions realm.

File: test/index.js (hunk 2 above)
Comment (maxbeatty) [inline test/index.js]: just to make sure the authorization header is not defined so it definitely goes to the first `Boom.unauthorized`. I can remove it if you prefer

File: test/index.js (hunk 1 above)
Comment (maxbeatty) [inline test/index.js]: :+1: 

off topic: what do you think about a separate PR to add `.eslintrc` with [hapi config](https://github.com/continuationlabs/eslint-plugin-hapi) and `.editorconfig` to match?

File: test/index.js (hunk 2 above)
Comment (mtharrison) [inline test/index.js]: Yes please.

File: test/index.js (hunk 1 above)
Comment (mtharrison) [inline test/index.js]: This is already built into [lab](https://github.com/hapijs/lab/tree/master/lib/linters/eslint). `npm test` runs the linter too.

Comment (maxbeatty) [inline test/index.js]: I was hoping lab defaulted to that. Doesn't look like it's catching the missing new line for the arrow function but correctly spots it for traditional function declaration. I'll open an issue with [hapi-scope-start](https://github.com/continuationlabs/hapi-scope-start)

Comment (mtharrison) [inline test/index.js]: It caught it for me. That's how I knew about it ;)

Comment (maxbeatty) [inline test/index.js]: Local issue :flushed:  `rm -r node_modules/ && npm i && npm test` caught it like it should have. found that `hapi-scope-start` has [a nice test](https://github.com/continuationlabs/hapi-scope-start/blob/master/test/index.js#L200) for exactly this

**Comment by mtharrison** [comment]
@maxbeatty would you be able to squash this into a single commit?

**Comment by maxbeatty** [comment]
Sure thing. What's the best way to do that? My git-foo is lacking

**Comment by mtharrison** [comment]
This guide might help: https://github.com/ginatrapani/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

**Comment by maxbeatty** [comment]
Thanks! That was much easier than the answers I was finding on SO

**Comment by mtharrison** [comment]
Thanks, published in v4.1.0.
